AWSTranscribeStreaming/AWSTranscribeStreamingService.m

```diff
@@ -223,32 +236,33 @@ - (void)startTranscriptionWSS:(AWSTranscribeStreamingStartStreamTranscriptionReq
                                                          code:AWSTranscribeStreamingClientErrorCodeWebSocketCouldNotInitialize
                                                      userInfo:@{NSUnderlyingErrorKey: error}];
 
-            [self.webSocketDelegateAdaptor webSocket:nil didFailWithError:wrappingError];
+            [self.webSocketProvider.delegate webSocket:self.webSocketProvider didError:wrappingError];
             return;
         }
 
-        if (!self.webSocket) {
+        if (!self.webSocketProvider) {
             error = [NSError errorWithDomain:AWSTranscribeStreamingClientErrorDomain
                                         code:AWSTranscribeStreamingClientErrorCodeWebSocketCouldNotInitialize
                                     userInfo:nil];
-            [self.webSocketDelegateAdaptor webSocket:nil didFailWithError:error];
+            [self.webSocketProvider.delegate webSocket:nil didError:error];
```

Comment:
it would be the exact same name as socket rockets protocol didFailWithError which would be confusing in the adaptor class?

Reply:
I don't have a particular problem with adaptors shadowing underlying methods especially since they so clearly represent the intent. In this case, we could if we wish rename to `-[webSocketProvider:didFailWithError:]` to be even more explicit.